installer.c

```diff
@@ -1094,13 +1104,46 @@ EFI_STATUS efi_main(EFI_HANDLE image, EFI_SYSTEM_TABLE *_table)
 	if (!include_self)
 		set_exclude_device(loaded_img->DeviceHandle);
 
+#ifdef USE_TPM
+	if (!is_live_boot() && is_platform_secure_boot_enabled()) {
+		ret = tpm2_init();
+		if (EFI_ERROR(ret) && ret != EFI_NOT_FOUND) {
+			efi_perror(ret, L"Failed to init TPM, exit");
+			goto exit;
+		}
+	}
+#endif  // USE_TPM
+
+	ret = set_device_security_info(NULL);
+	if (EFI_ERROR(ret)) {
+		efi_perror(ret, L"Failed to init security info, exit");
+		goto exit;
+	}
```

Comment:
I don't get it.  Usually `set_device_security_info`() returns `EFI_INVALID_PARAMETER` is the parameter is `NULL`.

Reply:
It is in ABL platform. But in UEFI platform, the parameter is usless.

Reply:
I see.  This is not a nice design though :/